pkg/register/restResolver.go

```diff
@@ -23,12 +23,47 @@ import (
 
 var defaultResolverTimeout = time.Second * 60
 
+// ResolverClientMetrics interface
+type ResolverClientMetrics interface {
+	RecordError(errType string)
```

Comment:
1. Add docstrings to say what these do (+ note about concurrency)
2. Turn `string` into own type so it's explicit what the possible values are